datameta/security/authz.py

```diff
@@ -16,18 +16,44 @@
 from ..models import MetaDatum, User
 
 
+def is_own_group(user, group):
+    return user.group.id == group.id
+
+
 def user_is_target(user, target_user):
     return user.id == target_user.id
 
 
 def has_group_rights(user, group):
-    return user.site_admin or (user.group_admin and user.group.id == group.id)
+    return user.site_admin or (user.group_admin and is_own_group(user, group))
 
 
 def is_power_grab(user, target_user):
     return not user.site_admin and target_user.site_admin
 
 
+def has_privileged_access(user):
+    return user.site_admin or user.site_read
```

Comment:
This function name in combination with `site_read` makes me nervous, it may be misinterpreted in the future. Maybe it's better not to encapsulate this OR clause for now but to write it down explicitly where needed.

Reply:
call it `has_privileged_read_access` instead?

Reply:
👍